static/app/components/codecov/container/codecovParamsProvider.tsx

```diff
@@ -1,74 +1,117 @@
     };
 
-    for (const [key, value] of Object.entries(validEntries)) {
-      if (!value || typeof value !== 'string') {
-        delete validEntries[key as keyof CodecovContextData];
+    for (const [key, value] of Object.entries(entries)) {
+      if (!value) {
+        delete entries[key as keyof typeof entries];
       }
     }
 
     setLocalStorageState(prev => ({
       ...prev,
-      ...validEntries,
+      ...entries,
     }));
-  }, [setLocalStorageState, location.query]);
+  }, [setLocalStorageState, searchParams]);
+
+  const changeContextValue = useCallback(
+    (value: Partial<CodecovContextDataParams>) => {
+      const currentParams = Object.fromEntries(searchParams.entries());
+      const valueKey = Object.keys(value)[0] as keyof typeof value;
+      const valuesToReset = VALUES_TO_RESET_MAP[valueKey];
+
+      for (const key of valuesToReset) {
+        delete currentParams[key];
+      }
+
+      setLocalStorageState((prev: Partial<CodecovContextDataParams>) => {
+        const newState = {...prev};
+        valuesToReset.forEach(key => {
+          delete newState[key as keyof CodecovContextDataParams];
+        });
+        return newState;
+      });
+
+      const updatedParams = {
+        ...currentParams,
+        ...value,
+      };
+
+      setSearchParams(updatedParams);
+    },
+    [searchParams, setLocalStorageState, setSearchParams]
+  );
 
   // Repository, org and branch default to null as its value to the option not being selected.
   // These only represent the unselected values and shouldn't be used when fetching backend data.
+  const queryRepository = searchParams.get('repository');
+  const queryIntegratedOrg = searchParams.get('integratedOrg');
+  const queryBranch = searchParams.get('branch');
+  const queryCodecovPeriod = searchParams.get('codecovPeriod');
+
   const params: CodecovContextData = {
-    repository:
-      typeof location.query.repository === 'string'
-        ? decodeURIComponent(location.query.repository)
-        : 'repository' in localStorageState
-          ? (localStorageState.repository as string)
-          : null,
-    integratedOrg:
-      typeof location.query.integratedOrg === 'string'
-        ? decodeURIComponent(location.query.integratedOrg)
-        : 'integratedOrg' in localStorageState
-          ? (localStorageState.integratedOrg as string)
-          : null,
-    branch:
-      typeof location.query.branch === 'string'
-        ? decodeURIComponent(location.query.branch)
-        : 'branch' in localStorageState
-          ? (localStorageState.branch as string)
-          : null,
-    codecovPeriod:
-      typeof location.query.codecovPeriod === 'string'
-        ? (decodeURIComponent(location.query.codecovPeriod) as CodecovPeriodOptions)
-        : 'codecovPeriod' in localStorageState
-          ? (localStorageState.codecovPeriod as CodecovPeriodOptions)
-          : '24h',
+    ...(typeof queryRepository === 'string'
```

Comment:
This is a reflection of updating the types originally. I had them like 

branch: string | null

Implying the key was always present in local storage, with string or null, but that isn't true. We either have the key or not, and if we have the key, it has a string, never a null, so updated the type to have optional keys with string, such as `branch?: string`